### PR TITLE
Implement throttling for WS client

### PR DESCRIPTION
**Pull request by Chengxuan**
The configurations already exist due to reusing ffresty config initialisation logic.

This PR implements the throttling logic itself

### CODE REVIEW

File: pkg/fftypes/uuid.go [hunk 1]
```diff
@@ -93,11 +93,13 @@ func (u *UUID) UnmarshalBinary(b []byte) error {
 }
 
 func (u *UUID) HashBucket(buckets int) int {
-	if u == nil {
+	if u == nil || buckets <= 0 {
```
Comment (EnriqueL8) [inline pkg/fftypes/uuid.go:96]: why? the next line would NPE if buckets > 0

File: pkg/ffresty/ffresty.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ type Config struct {
 }
 
 var (
-	rateLimiter    *rate.Limiter
+	rateLimiterMap map[*resty.Client]*rate.Limiter
```
Comment (EnriqueL8) [inline pkg/ffresty/ffresty.go:57]: confused about this one.. we are using a local variable so where we overriding this rate limiter each time we created a new client and you are fixing this bug?

File: pkg/ffresty/ffresty.go [hunk 3]
```diff
@@ -242,9 +245,9 @@ func NewWithConfig(ctx context.Context, ffrestyConfig Config) (client *resty.Cli
 	client.SetTimeout(time.Duration(ffrestyConfig.HTTPRequestTimeout))
 
 	client.OnBeforeRequest(func(_ *resty.Client, req *resty.Request) error {
-		if rateLimiter != nil {
+		if rateLimiterMap[client] != nil {
```
Comment (EnriqueL8) [inline pkg/ffresty/ffresty.go:248]: I wonder if the rate limiter can be a property of the resty client ... seems like it is there https://github.com/go-resty/resty/pull/715/files

Comment (EnriqueL8) [reply pkg/ffresty/ffresty.go:248]: ah the way is implemented is different, we wait and the implementation of resty throws an error

**Review by EnriqueL8 — CHANGES_REQUESTED** [review]
Looking good - a few questions

File: pkg/ffresty/ffresty.go (hunk 2 above)
Comment (Chengxuan) [reply pkg/ffresty/ffresty.go:57]: yes this is a bug fix

File: pkg/ffresty/ffresty.go (hunk 3 above)
Comment (Chengxuan) [reply pkg/ffresty/ffresty.go:248]: i think this is resolved

File: pkg/fftypes/uuid.go (hunk 1 above)
Comment (Chengxuan) [reply pkg/fftypes/uuid.go:96]: converting a negative integer to uint doesn't make sense.

Comment (EnriqueL8) [reply pkg/fftypes/uuid.go:96]: ah sorry yeah makes sense

**Review by EnriqueL8 — APPROVED** [review]
Looks good

**Pull request merged**